### PR TITLE
pkg/*: switch to v1 CRD client

### PR DESCRIPTION
**Pull request by s-urbaniak**
https://github.com/kubernetes-sigs/controller-tools/pull/356 added support for generating v1 CRDs, hence it is time to migrate here too to enable pruning by default. I tested that updating CRDs does no harm.

cc @LiliC @brancz @pgier @metalmatze @squat

### CODE REVIEW

**Comment by s-urbaniak** [comment]
(let's see what CI and e2e say)

**Comment by s-urbaniak** [comment]
from what I see a bunch of mechanical failures, i'll look into them.

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ VERSION?=$(shell cat VERSION | tr -d " \t\n\r")
 
 FIRST_GOPATH:=$(firstword $(subst :, ,$(shell go env GOPATH)))
 CONTROLLER_GEN_BINARY := $(FIRST_GOPATH)/bin/controller-gen
-CRD_OPTIONS ?= "crd:preserveUnknownFields=false"
+CRD_OPTIONS ?= "crd:crdVersions=v1,preserveUnknownFields=false"
```
Comment (pgier) [inline Makefile]: `preserveUnknownFields=false` is the default in v1, so it should be ok to remove that from the setting if we want.

Comment (s-urbaniak) [reply Makefile]: indeed, good catch :+1:

**Comment by s-urbaniak** [comment]
thanks @LiliC for peeking! let's handle this PR with lower prio though right now. I just wanted to estimate how much of an effort this would be.

**Comment by pgier** [comment]
One risk of this change is that it would make us dependent on kubernetes version 1.16.0 or higher. Prior to that version, the CRD was only in the v1beta1 api.  So attempting to deploy the CRDs on 1.15.0 for example, you get errors like this:
 `error: unable to recognize "example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1"`

**Comment by s-urbaniak** [comment]
@pgier agreed :+1: we have to weigh if this is important enough to move forward. Having said that we are in pre-1.0 release state.

**Comment by paulfantom** [comment]
@pgier @s-urbaniak since k8s 1.16 is now the minimal supported version as per [kubernetes version skew policy](https://kubernetes.io/docs/setup/release/version-skew-policy/) can we go with this?

**Comment by simonpasquier** [comment]
I guess it can be closed now since we've switched to CRD v1.

**Pull request closed without merging**